libcxx/include/experimental/__simd/simd.h

```diff
@@ -26,15 +27,29 @@
 _LIBCPP_BEGIN_NAMESPACE_EXPERIMENTAL
 inline namespace parallelism_v2 {
 
+template <class _Simd, class _Impl, bool>
+class __simd_int_operators {};
+
+template <class _Simd, class _Impl>
+class __simd_int_operators<_Simd, _Impl, true> {
+public:
+  // unary operators for integral _Tp
+  _LIBCPP_HIDE_FROM_ABI _Simd operator~() const noexcept {
```

Comment:
Couldn't we also just `enable_if` this individually?

Reply:
Thanks for the suggestion! I opted for `__simd_int_operators` because I need to add more integer-only operators (like bitwise, modulus, shifts and their compound assignments) later and thought centralizing them here would keep things consistent. Do you think it’s better to stick with this, or switch to using `enable_if_t` individually for each operator?

Reply:
If we need this for a bunch of operators I think it makes sense to keep the base class.